lib/data/entity.js

```diff
@@ -380,13 +392,35 @@ const getWithConflictDetails = (defs) => {
       if ('label' in v.dataReceived && v.dataReceived.label !== defMap.get(v.version - 1).label) v.serverDiff.push('label');
 
       if (conflict) {
-        v.conflict = v.conflictingProp && v.conflictingProp.length > 0 ? ConflictType.HARD : ConflictType.SOFT;
+        v.conflict = v.conflictingProperties && v.conflictingProperties.length > 0 ? ConflictType.HARD : ConflictType.SOFT;
+
+        v.resolved = lastResolveEvent && lastResolveEvent.details.entityDefId >= def.id;
+
+        // lastGoodVersion is the baseVersion of first unresolved conflict
+        if (!v.resolved && lastGoodVersion === 0) {
+          lastGoodVersion = v.baseVersion;
```

Comment:
The release criteria define it as:

> The update immediately prior to the earliest parallel update (hereafter the “known good” record)

So I think:

```diff
-lastGoodVersion = v.baseVersion;
+lastGoodVersion = v.version - 1;
```

Reply:
`v.version - 1` is parallel to `v` on which we have marked the conflict. Although, `v.version - 1` has no conflict in the database because it arrived before `v`, it may still not be the truth. Both versions are in contention.

Consider following scenario, I think v30 is the last known good version, because (31, 32) and (33) are two parallel update branches

```

                            ┌───────────┐            ┌──────────┐
           ┌───────────────►│           │            │          │
           │                │    v31    ├───────────►│   v32    │
           │                │           │            │          │
           │                └───────────┘            └──────────┘
           │
   ┌───────┴──────┐
   │              │
   │  v30         │
   │              │
   └────────┬─────┘
            │
            │
            │
            │
            │                                                     ┌───────────┐
            │                                                     │           │
            └─────────────────────────────────────────────────────►   v33     │
                                                                  │           │
                                                                  └───────────┘

```

Reply:
If v32 were considered the last good version, then we would show v32, v33 (the conflict), and v30 (the base version of the conflict) in the summary table. We'd be skipping v31, but I'm not sure that the user needs to compare v31 and v33, as v32 includes the changes from v31. Maybe in some cases the user would want to thoroughly review the history, but I'm not sure that we need to include v31 in the summary table.

I do see what you're saying about the two branches. I think taking that approach would have design implications though. For example, what status icon would be shown in the summary table for v31? I don't think the current icons and text would work. What do you think about starting with `v.version - 1`, then we can ask issa what she thinks? If you don't like the language of "last good", we could use a more specific term ("last resolved"? "last before conflict"?).

By the way, if we were to account for branching, I think we'd need to adjust the current approach slightly. In that case, I think the last good version wouldn't be the base version of the first unresolved conflict; it would be the minimum base version among unresolved conflicts. The second conflict might have an earlier base version than the first conflict.